tracks/python/exercises/markdown/README.md

```diff
@@ -0,0 +1,90 @@
+### Problem and challenges
+	
+The markdown exercise asks the student to refactor some code. The original code is 
+hard to follow; it is constructed of a single `for` loop, has many nested 
+conditionals, and meaningless variable names. The goal is for them to produce 
+improved code that still passes the tests.
+	
+### Reasonable solution
+	
+```python
+import re
+
+
+def format_header(line):
+    def replacement(match):
+        hashes, text = match.groups()
+        num = len(hashes)
+        return f'<h{num}>{text}</h{num}>'
+    line = re.sub(r'^(#{1,6}) (.*)', replacement, line)
+    return line
+
+
+def format_bold(curr):
+    curr = re.sub(r'__(.*?)__', r'<strong>\1</strong>', curr)
```

Comment:
Same here, you're allocating **curr** to shadow **curr** from the outer scope ... just `return re.sub([...])`.